homework/index.js

```diff
@@ -29,23 +24,149 @@
+      href: repo.html_url,
+      target: '_blank',
+    });
+
+    if (repo.description === null) {
+      createAndAppend('span', description, {
+        text: repo.description || 'No Description Added Yet',
+        class: 'texts',
+      });
+    }
+
+    createAndAppend('span', forkedNumbers, {
+      text: repo.forks,
+      class: 'texts',
+    });
+
+    createAndAppend('span', updatedTime, {
+      text: convertDateTime(repo.updated_at),
+      class: 'texts',
+    });
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
-        });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+  function renderContributor(contributor, ul) {
+    const li = createAndAppend('li', ul, { class: 'contributors-li' });
+    const contributorImage = createAndAppend('div', li, {
+      class: 'contributor-image',
+    });
+    createAndAppend('img', contributorImage, {
+      src: contributor.avatar_url,
+      alt: contributor.login,
     });
+
+    const contributorName = createAndAppend('div', li, {
+      class: 'contributor-name',
+    });
+    createAndAppend('a', contributorName, {
+      text: contributor.login,
+      href: contributor.html_url,
+      target: '_blank',
+    });
+
+    createAndAppend('span', contributorName, {
+      text: contributor.contributions,
+      class: 'added-contributions',
+    });
+  }
+
+  async function getContributorsInfo(repositories, ul) {
+    const url = repositories.contributors_url;
+    try {
+      const contributors = await fetchJSON(url);
+      ul.innerHTML = '';
+      contributors.forEach(contributor => renderContributor(contributor, ul));
+    } catch (err) {
+      createAndAppend('div', document.getElementById('root'), {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  async function main(url) {
+    try {
+      const root = document.getElementById('root');
+      const headerDiv = createAndAppend('header', root, {
+        text: 'HYF Repositories',
+        class: 'repo-header',
+      });
+      const select = createAndAppend('select', headerDiv, {
+        class: 'select-repository',
+      });
+      const repos = await fetchJSON(url);
+      repos.sort(sortRepositoriesByName).forEach((repo, index) =>
+        createAndAppend('option', select, {
+          text: repo.name,
+          value: index,
+        }),
+      );
+      const mainContainer = createAndAppend('main', root, {
+        class: 'main-container',
+      });
+      const repositoryContainer = createAndAppend('section', mainContainer, {
+        class: 'repository-section',
+      });
+      const ul = createAndAppend('ul', repositoryContainer, {
+        class: 'ul',
+      });
+      renderRepoDetails(repos[0], ul);
```

Comment:
For consistency and improved readability, place this statement near line 157, so that it becomes immediately obvious that the same code is executed as you do in the event handler:

```js
renderRepoDetails(repos[0], ul);
getContributorsInfo(repos[0], ulContributors);

select.addEventListener('change', () => {
  const repo = repos[select.value];
  renderRepoDetails(repo, ul);
  getContributorsInfo(repo, ulContributors);
});
```

Reply:
i change the code as needed, thank you for the feedback